src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -118,12 +118,19 @@ BlockIO InterpreterCreateUserQuery::execute()
                 access->checkAdminOption(role);
         }
     }
-    if (!query.cluster.empty())
-        return executeDDLQueryOnCluster(query_ptr, getContext());
+
     std::optional<SettingsProfileElements> settings_from_query;
     if (query.settings)
+    {
         settings_from_query = SettingsProfileElements{*query.settings, access_control};
 
+        if (!query.attach)
+            getContext()->checkSettingsConstraints(*settings_from_query);
```

Comment:
This is good, but I think for `CREATE / ALTER USER` we also need to add all the current constraints silently in the code to the statement unless they're overridden explicitly by stricter constraints.

For example after executing

```
CREATE USER vasya;
```

the user `vasya` should have the current constraints (not no constraints).

Reply:
Yes, you're right. But we can silently inherit the current profiles too. Maybe not profiles as a whole, but only their constraints. Something like this:

```
CREATE USER vasya SETTINGS x=8, CONSTRAINTS FROM profile1, CONSTRAINTS FROM profile2, y=9 MIN 5 MAX 10
```